plugin/src/test/java/hudson/plugins/swarm/test/TestUtils.java

```diff
@@ -0,0 +1,92 @@
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import com.google.common.io.ByteStreams;
+import hudson.Util;
+import hudson.model.Computer;
+import hudson.model.Node;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URL;
+import jenkins.model.Jenkins;
+import org.jvnet.hudson.test.JenkinsRule;
+
+/** Utilities for testing the Swarm Plugin */
+public class TestUtils {
+
+    /** Download the Swarm Client from the given Jenkins URL into the given temporary directory. */
+    private static File download(URL jenkinsUrl, File tmpDir) throws Exception {
+        assertTrue(tmpDir.isDirectory());
+
+        URL input =
+                jenkinsUrl.toURI().resolve(new URI(null, "swarm/swarm-client.jar", null)).toURL();
+        File output = new File(tmpDir, "swarm-client.jar");
+        try (InputStream inputStream = input.openStream();
+                FileOutputStream outputStream = new FileOutputStream(output)) {
+            ByteStreams.copy(inputStream, outputStream);
+        }
+
+        assertTrue(output.isFile());
+        assertThat(output.length(), greaterThan(1000L));
+        return output;
+    }
+
+    /** Wait for the agent with the given name to come online against the given Jenkins instance. */
+    private static Computer waitOnline(String agentName, Jenkins jenkins)
+            throws InterruptedException, IOException {
+        Computer computer = jenkins.getComputer(agentName);
+        while (computer == null) {
+            Thread.sleep(100);
+            computer = jenkins.getComputer(agentName);
+        }
+
+        while (!computer.isOnline()) {
+            Thread.sleep(100);
+        }
+
+        return computer;
+    }
+
+    /**
+     * Create a new Swarm Client agent on the local host and wait for it to come online before
+     * returning.
+     */
+    public static Node createSwarmClient(
+            JenkinsRule j, ProcessDestroyer processDestroyer, File tmpDir, String... args)
+            throws Exception {
+        String agentName = "agent" + j.jenkins.getNodes().size();
+        File swarmClientJar = download(j.getURL(), tmpDir);
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("\"" + System.getProperty("java.home") + "/bin/java\"");
+        sb.append(" -Djava.awt.headless=true");
+        sb.append(" -jar \"" + swarmClientJar.getAbsolutePath() + "\"");
+        sb.append(" -name \"" + agentName + "\"");
+        sb.append(" -master \"" + j.getURL() + "\"");
+        // "-disableClientsUniqueId" is needed so that we can call getComputer() in the waitOnline()
+        // method.
+        sb.append(" -disableClientsUniqueId");
+        for (String arg : args) {
+            sb.append(" " + arg);
+        }
+
+        ProcessBuilder pb = new ProcessBuilder(Util.tokenize(sb.toString()));
```

Comment:
So, simpler to construct a `List<String> argv` to begin with, and dispense with the quoting.

Reply:
Done.